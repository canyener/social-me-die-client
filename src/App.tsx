import React, { Component } from 'react'
import { Header, Icon, List } from 'semantic-ui-react'

import './App.css'
import axios from 'axios'

class App extends Component {
  state = {
    values: []
  }

  componentDidMount() {
    axios.get('http://localhost:5000/api/values').then(response => {
      this.setState(() => ({ values: response.data }))
    })
  }

  render() {
    return (
      <div>
        <Header as='h2'>
          <Icon name='users' />
          <Header.Content>Social Me Die</Header.Content>
        </Header>
        <List>
          {
            this.state.values.map((value: any) => 
              <List.Item key={value.id}>
                {value.name}
              </List.Item>)
          }
        </List>
      </div>
    )
  }
}

export default App
