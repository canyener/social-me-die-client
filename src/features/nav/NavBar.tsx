import React from 'react'
import { Menu, Container, Button } from 'semantic-ui-react'

export const NavBar = () => (
  <Menu fixed='top' inverted>
    <Container>
      <Menu.Item header>
        <img src="/assets/logo.png" alt="logo" style={{ marginRight: '10px' }} />
          Social Me Die
        </Menu.Item>
      <Menu.Item name='Activities' />
      <Menu.Item>
        <Button positive content='Create Activity' />
      </Menu.Item>
    </Container>
  </Menu>
)
