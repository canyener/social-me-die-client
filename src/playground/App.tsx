import React, { Component } from 'react'
import { Header, Icon, List } from 'semantic-ui-react'

import axios from 'axios'

import { IActivity } from '../app/models/activity'

interface IState {
  activities: IActivity[]
}

class App extends Component<{}, IState> {
  readonly state : IState = {
    activities: []
  }

  componentDidMount() {
    axios.get<IActivity[]>('http://localhost:5000/api/activities').then(response => {
      this.setState(() => ({ 
        activities: response.data 
      }))
    })
  }

  render() {
    return (
      <div>
        <Header as='h2'>
          <Icon name='users' />
          <Header.Content>Social Me Die</Header.Content>
        </Header>
        <List>
          {
            this.state.activities.map((activity) => 
              <List.Item key={activity.id}>
                {activity.title}
              </List.Item>)
          }
        </List>
      </div>
    )
  }
}

export default App
