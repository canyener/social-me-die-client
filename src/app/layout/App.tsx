import React, { useState, useEffect, Fragment } from 'react'
import { List, Container } from 'semantic-ui-react'

import axios from 'axios'

import { IActivity } from '../models/activity'
import { NavBar } from '../../features/nav/NavBar'

const App = () => {

  const [activities, setActivities] = useState<IActivity[]>([])

  useEffect(() => {
    axios.get<IActivity[]>('http://localhost:5000/api/activities').then(response => {
      setActivities(response.data)
    })
  }, [])

  return (
    <Fragment>
      <NavBar />
      <Container style={{ marginTop: '7em' }}>
        <List>
          {
            activities.map((activity) =>
              <List.Item key={activity.id}>
                {activity.title}
              </List.Item>)
          }
        </List>
      </Container>
    </Fragment>
  )
}

export default App
