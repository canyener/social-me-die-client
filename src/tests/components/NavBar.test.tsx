import React from 'react'
import { shallow, mount } from 'enzyme'

import { NavBar } from '../../features/nav/NavBar'
import App from '../../app/layout/App'

test('Should render NavBar correctly', () => {
  const wrapper = shallow(<NavBar openCreateForm={()=> {}} />)

  expect(wrapper).toMatchSnapshot()
})

test('Should call handler for openCreateForm when Create Activity button is clicked', () => {

  const openCreateFormSpy = jest.fn()
  
  const wrapper = shallow(<NavBar openCreateForm={openCreateFormSpy} />)
  wrapper.find('Button').first().simulate('click')
  expect(openCreateFormSpy).toHaveBeenCalledTimes(1)
})