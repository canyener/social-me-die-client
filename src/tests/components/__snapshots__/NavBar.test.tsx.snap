// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render NavBar correctly 1`] = `
<Menu
  fixed="top"
  inverted={true}
>
  <Container>
    <MenuItem
      header={true}
    >
      <img
        alt="logo"
        src="/assets/logo.png"
        style={
          Object {
            "marginRight": "10px",
          }
        }
      />
      Social Me Die
    </MenuItem>
    <MenuItem
      name="Activities"
    />
    <MenuItem>
      <Button
        as="button"
        content="Create Activity"
        positive={true}
      />
    </MenuItem>
  </Container>
</Menu>
`;
